source/auth/auth.md

```diff
@@ -1201,15 +1201,6 @@ in the MONGODB-OIDC specification, including sections or blocks that specificall
 
 #### [MongoCredential](#mongocredential) Properties
 
-> [!NOTE]
-> Drivers MUST NOT url-decode the entire `authMechanismProperties` given in an connection string when the
-> `authMechanism` is `MONGODB-OIDC`. This is because the `TOKEN_RESOURCE` itself will typically be a URL and may contain
-> a `,` character. The values of the individual `authMechanismProperties` MUST still be url-decoded when given as part
-> of the connection string, and MUST NOT be url-decoded when not given as part of the connection string, such as through
-> a `MongoClient` or `Credential` property. Drivers MUST parse the `TOKEN_RESOURCE` by splitting only on the first `:`
-> character. Drivers MUST document that users must url-encode `TOKEN_RESOURCE` when it is provided in the connection
-> string and it contains and of the special characters in \[`,`, `+`, `&`, `%`\].
```

Comment:
I think we should still specify that this should be documented. Note that certain other characters still affect parsing of TOKEN_RESOURCE. It would be helpful to write the doc requirement in a way that can be copy-pasted into individual drivers:

> Drivers MUST document that: If the value of TOKEN_RESOURCE contains the characters \[`,`, `+`, `&`, `%`\], then it must not be specified in the connection string, but must instead be specified via a `MongoClient` option.

We can also say the following to allow more characters in the connection string, but it seems more convoluted:

> Drivers MUST document that: If the value of TOKEN_RESOURCE contains the characters \[`+`, `&`, `%`\], then it must be percent-encoded when specified in the connection string. If the value of TOKEN_RESOURCE contains a comma \[`,`\], then it must not be specified in the connection string, but must instead be specified via a `MongoClient` option.

Reply:
I don't think we need to call out TOKEN_RESOURCE as special, but keep the logic in the connection string file.

Reply:
I do not mean the parsing logic, referred to in an [earlier comment](https://github.com/mongodb/specifications/pull/1588#discussion_r1619009956), which I agree need not be specified individually.

I mean that we previously specified that drivers must document this non-obvious behaviour in user-facing documentation, in at least one location. We required this on TOKEN_RESOURCE because it was apparently the first case where it became relevant. Drivers should be as careful in how they document behaviour contracts, as they are in the implementation, since breaking the contract is what will be perceived by users as breaking compatibility.

I do think that this should be documented individually on each key-value-pair affected by this rule (especially since I believe TOKEN_RESOURCE is the only one affected).